### PR TITLE
LA-222: Change detail Id to be integer and address sonar warning

### PR DESCRIPTION
**Pull request by bpoon-selina**

### CODE REVIEW

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **bpoon-selina**
You can assign the PR to them by writing `/assign @bpoon-selina` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-111** [here](http://ms-lending-api-pr111.selinadev.co.uk)

File: src/test/java/com/selina/lending/LendingServiceApplicationTests.java [hunk 1]
```diff
@@ -17,13 +17,37 @@
...
 
+import com.selina.lending.api.controller.CreditCommitmentsController;
+import com.selina.lending.api.controller.DIPController;
+import com.selina.lending.api.controller.ProductController;
+import com.selina.lending.api.controller.QuickQuoteController;
+
 @SpringBootTest
 class LendingServiceApplicationTests {
 
+	@Autowired
```
Comment (Vadym-Lopatka) [inline src/test/java/com/selina/lending/LendingServiceApplicationTests.java:34]: Small question, why do you do these additional asserts about not being null?

As I see it, in this type of test that runs all the Spring context - we definitely should fail if some bean was injected but was not defined.

**Review by Vadym-Lopatka — CHANGES_REQUESTED** [review]

File: src/test/java/com/selina/lending/LendingServiceApplicationTests.java (hunk 1 above)
Comment (bpoon-selina) [reply src/test/java/com/selina/lending/LendingServiceApplicationTests.java]: added NOSONAR instead of assertions

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-111** [here](http://ms-lending-api-pr111.selinadev.co.uk)

**Pull request merged**